src/docfx/restore/FileResolver.cs

```diff
@@ -83,7 +87,7 @@ public string ResolveFilePath(SourceInfo<string> file)
 
         public async Task Download(SourceInfo<string> file)
         {
-            if (!UrlUtility.IsHttp(file))
+            if (!UrlUtility.IsHttp(file) || s_downloadedUrls.Contains(file))
```

Comment:
There is still chance to download more than once, right?

Reply:
yes, tiny.

Reply:
ok, fair trade off to achieve better perf without lock